controller/utils.go

```diff
@@ -157,55 +233,46 @@ func createNewRemoveOldPod(
 	return false, false, nil
 }
 
-// In rollback, it must delete newly created pod and
-// restore old deleted pods to come back to previous state
-func rollback(
+func dbRollback(
 	ctx context.Context,
 	logger logrus.FieldLogger,
-	roomManager models.RoomManager,
 	mr *models.MixedMetricsReporter,
 	db pginterfaces.DB,
 	redisClient *redis.Client,
-	clientset kubernetes.Interface,
 	failedConfigYAML *models.ConfigYAML,
 	oldConfigYAML *models.ConfigYAML,
-	maxSurge int,
 	clock clockinterfaces.Clock,
 	scheduler *models.Scheduler,
 	config *viper.Viper,
 	operationManager *models.OperationManager,
 	oldVersion, globalLockKey string,
-) error {
-	schedulerName := oldConfigYAML.Name
-	l := logger.WithFields(logrus.Fields{
-		"source":    "updateSchedulerConfig.rollback",
-		"scheduler": schedulerName,
-	})
-
-	l.Debug("starting rollback")
-
-	globalLock, _, _ := acquireLock(
+) (err error) {
+	globalLock, _, _ := AcquireLock(
```

Comment:
globalLock could be nil, I think you need to check err. Don't you also need to check canceled?

Reply:
I actually don't need to global lock here anymore. dbRollback runs in UpdateSchedulerConfig, which already locks config (can't do more than 1 update at the same time).

EnsureCorrectRooms is not using global lock as well (it uses downscaling lock)